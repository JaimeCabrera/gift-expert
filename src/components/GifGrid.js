import React from 'react'
import PropTypes from 'prop-types'
import GifGridItem from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

const GifGrid = ({ category }) => {

   // para renombrar en la desetructuracion name:newname
   const {data:images, loading} = useFetchGifs(category);
   // const [images, setImages] = useState([]);

   // useEffect(() => {
   //    getGifs(category).then(gifs => setImages(gifs));
   // }, [category])

   // recibimos la categoria y buscamos los gifs de esa categoria
   
   return (
      <>
         <h3 className="animate__animated animate__fadeIn">{category}</h3>
         {loading && <p className="animate__animated animate__flash">Cargando...</p>}
         <div className="card-grid">
            {
               images.map(img => {
                  // return <li key={id}>{title}</li>
                  return <GifGridItem
                     key={img.id}
                     {...img}
                  />
               })
            }

         </div>
      </>
   )
}

GifGrid.propTypes = {
   category:PropTypes.string
}

export default GifGrid
