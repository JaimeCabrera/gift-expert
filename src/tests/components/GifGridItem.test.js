import React from 'react'

import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import {GifGridItem} from '../../components/GifGridItem';

describe('Pruebas en el componente GifGridItem.js', () => {

   const title = "esto es un titulo";
   const url = "https://imagenes/img.jpg";
   const wrapper = shallow(<GifGridItem title={title} url={url} />);

   test('debe renderizar el componente <GifGridItem/> correctament', () => {
      expect(wrapper).toMatchSnapshot();
   });

   test('Debe de mostrar un parrafo {p} con el titulo ', () => {
      const p = wrapper.find('p');
      expect(p.text().trim()).toBe(title);
   });

   test('debe de mostrar una imagen {img} con el url y title del props ', () => {
      const img = wrapper.find('img');
      // para ver los atributos seria img.html() pero si uso img.props se ve un obj de propiedades
      // si solo se quirer evaluar una propiedad seria img.prop('src)
      expect(img.props().src).toBe(url);
      expect(img.props().alt).toBe(title);
   });

   test('Debe el div principal debe de tener la clase animate__fadeIn ', () => {
      const div = wrapper.find('div');
      const clasName = div.props().className;
      expect(clasName.includes('animate__fadeIn')).toBe(true);
   })
   
   
   
})