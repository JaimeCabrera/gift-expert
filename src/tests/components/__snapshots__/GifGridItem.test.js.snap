// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGridItem.js debe renderizar el componente <GifGridItem/> correctament 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="esto es un titulo"
    src="https://imagenes/img.jpg"
  />
  <p>
    esto es un titulo
  </p>
</div>
`;
